### PR TITLE
[Galakhtic] Updates for Habitat 2.0 Bridge

### PR DESCRIPTION
**Pull request by ASzot**
## Motivation and Context

Some changes were necessary to train policies correctly so they can be properly imported into Habitat 2.0. 

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 1]
```diff
@@ -77,25 +78,32 @@ def __init__(
...
 
     @classmethod
     def from_config(
-        cls, config: Config, observation_space: spaces.Dict, action_space
+        cls,
+        config: Config,
+        observation_space: spaces.Dict,
+        action_space,
     ):
+        any_rgb = any(["rgb" in k for k in observation_space.spaces])
```
Comment (Skylion007) [inline habitat_baselines/rl/ddppo/policy/resnet_policy.py:95]:
```suggestion
        any_rgb = any("rgb" in k for k in observation_space.spaces)
```
Any takes a generator.

File: habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 2]
```diff
@@ -111,17 +119,16 @@ def __init__(
...
 
-        if "depth" in observation_space.spaces:
-            self._n_input_depth = observation_space.spaces["depth"].shape[2]
-            spatial_size = observation_space.spaces["depth"].shape[0] // 2
-        else:
-            self._n_input_depth = 0
+        # Count total # of channels for rgb and for depth
+        self._n_input_rgb, self._n_input_depth = [
+            # sum() returns 0 for an empty list
+            sum([observation_space.spaces[k].shape[2] for k in keys])
```
Comment (Skylion007) [inline habitat_baselines/rl/ddppo/policy/resnet_policy.py:129]:
```suggestion
            sum(observation_space.spaces[k].shape[2] for k in keys)
```

File: habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 3]
```diff
@@ -223,18 +240,32 @@ def __init__(
...
+        # Only fuse the 1D state inputs. Other inputs are processed by the
+        # visual encoder
+        if fuse_keys is None:
+            fuse_keys = []
+        self._fuse_keys: List[str] = [
+            k for k in fuse_keys if len(observation_space.spaces[k].shape) == 1
+        ]
+        if len(self._fuse_keys) != 0:
+            rnn_input_size += sum(
+                [observation_space.spaces[k].shape[0] for k in self._fuse_keys]
```
Comment (Skylion007) [inline habitat_baselines/rl/ddppo/policy/resnet_policy.py:267]:
```suggestion
                (observation_space.spaces[k].shape[0] for k in self._fuse_keys)
```

File: habitat_baselines/utils/common.py [hunk 4]
```diff
@@ -594,3 +630,72 @@ def action_to_velocity_control(
...
+                queue.append(v)
+        elif isinstance(v, spaces.Box):
+            num_actions += v.shape[0]
+        else:
+            num_actions += 1
+
+    return num_actions
+
+
+def action_array_to_dict(
```
Comment (eundersander) [inline habitat_baselines/utils/common.py:664]: I don't understand what this is. I don't see it used anywhere. I hope it's not on a perf-critical path like running after every DNN inference for all envs. In my mind, actions are just a big flat list of floats coming out of a DNN and piped directly into the Gala sim, so I'm surprised at all this complexity.

**Review by eundersander — COMMENTED** [review]
I skimmed this and left a few thoughts.

File: habitat_baselines/config/rearrange/gala_kinematic_ddppo.yaml [hunk 5]
```diff
@@ -1,5 +1,8 @@
 TENSORBOARD_DIR: "/checkpoint/eundersander/gala_kinematic/tb/gala_kinematic_ddppo"
 CHECKPOINT_FOLDER: "/checkpoint/eundersander/gala_kinematic/ckpt/gala_kinematic_ddppo"
+#EPISODES_FILE: "data/datasets/train/all_receptacles_10k_1k.json"
+EPISODES_FILE: "generated.episode_set.json"
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat_baselines/config/rearrange/gala_kinematic_ddppo.yaml:4]: No, not yet. The episode dataset is not working yet. That is why I kept `PROCEDURAL_EPISODE_GENERATION: True`. Once it is ready I can add it though.

File: habitat_baselines/rl/ddppo/policy/resnet_policy_bpsnav.py [hunk 6]
```diff
@@ -321,16 +348,29 @@ def __init__(
...
         self._n_prev_action = 32
         rnn_input_size = self._n_prev_action
 
-        if "pointgoal_with_gps_compass" in observation_space.spaces:
-            n_input_goal = 3
-            self.tgt_embeding = nn.Linear(n_input_goal, 32)
-            rnn_input_size += 32
+        self._include_inputs: List[str] = [
+            k
+            for k in include_inputs
```
Comment (vincentpierre) [inline habitat_baselines/rl/ddppo/policy/resnet_policy_bpsnav.py:359]:
```suggestion
            for k in observation_space.spaces
```
We already have a list of sensors we want to include in gala_kinematic_ddppo.yaml

**Pull request closed without merging**